x-pack/plugins/observability_solution/observability_logs_explorer/public/components/alerts_popover.tsx

```diff
@@ -116,7 +129,7 @@ export const AlertsPopover = () => {
           params: {
             searchConfiguration: {
               index,
-              query: logsExplorerState.query,
```

Comment:
@maryam-saeidi I am just wondering, why they didn't have to cast `as Query` similar to how we were doing in custom threshold?

Reply:
Because previously, we were accepting query as `Query | AggregateQuery` and we were casting it on our side, which was dangerous, as if the query was passed as `AggregateQuery`, didn't have any way to know it. (Although, I didn't find a way to break it because `AggregateQuery` is not handled on the Logs Explorer page yet.)

Now, I am ensuring that the type is exactly what the component needs and the consumer can decide what to do with the other types which in this case, I am just ignoring it for now.

Reply:
Thanks makes sense!